server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -8,41 +8,49 @@
 
 package org.elasticsearch.index.mapper.vectors;
 
+import org.apache.lucene.document.FeatureField;
 import org.apache.lucene.search.Query;
+import org.apache.lucene.util.BytesRef;
 import org.elasticsearch.common.logging.DeprecationCategory;
-import org.elasticsearch.common.logging.DeprecationLogger;
+import org.elasticsearch.common.lucene.Lucene;
 import org.elasticsearch.index.IndexVersion;
+import org.elasticsearch.index.analysis.NamedAnalyzer;
+import org.elasticsearch.index.fielddata.FieldDataContext;
+import org.elasticsearch.index.fielddata.IndexFieldData;
 import org.elasticsearch.index.mapper.DocumentParserContext;
 import org.elasticsearch.index.mapper.FieldMapper;
 import org.elasticsearch.index.mapper.MappedFieldType;
 import org.elasticsearch.index.mapper.MapperBuilderContext;
+import org.elasticsearch.index.mapper.SourceValueFetcher;
 import org.elasticsearch.index.mapper.TextSearchInfo;
 import org.elasticsearch.index.mapper.ValueFetcher;
 import org.elasticsearch.index.query.SearchExecutionContext;
-import org.elasticsearch.search.DocValueFormat;
+import org.elasticsearch.xcontent.XContentParser.Token;
 
-import java.time.ZoneId;
+import java.io.IOException;
 import java.util.Map;
 
+import static org.elasticsearch.index.query.AbstractQueryBuilder.DEFAULT_BOOST;
+
 /**
- * A {@link FieldMapper} for indexing a sparse vector of floats.
- *
- * @deprecated The sparse_vector type was deprecated in 7.x and removed in 8.0. This mapper
- * definition only exists so that 7.x indices can be read without error.
- *
- * TODO: remove in 9.0.
+ * A {@link FieldMapper} that exposes Lucene's {@link FeatureField} as a sparse
+ * vector of features.
  */
-@Deprecated
 public class SparseVectorFieldMapper extends FieldMapper {
-    private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(SparseVectorFieldMapper.class);
-    static final String ERROR_MESSAGE = "The [sparse_vector] field type is no longer supported.";
-    static final String ERROR_MESSAGE_7X = "The [sparse_vector] field type is no longer supported. Old 7.x indices are allowed to "
-        + "contain [sparse_vector] fields, but they cannot be indexed or searched.";
+
     public static final String CONTENT_TYPE = "sparse_vector";
 
+    static final String ERROR_MESSAGE_7X = "[sparse_vector] field type in old 7.x indices is allowed to "
```

Comment:
Nitpicky question: Should we use the `CONTENT_TYPE` constant in error messages instead of hardcoding `sparse_vector`?

Reply:
I copied / pasted from the previous `sparse_vector` error messages. As content type constant is unlikely to change, I personally think this is more readable as well (and in line with how error messages are created as well)